lib/membrane_rtmp_plugin/rtmp/source/bin.ex

```diff
@@ -21,34 +22,18 @@ defmodule Membrane.RTMP.SourceBin do
     mode: :pull,
     demand_unit: :buffers
 
-  def_options port: [
-                spec: 1..65_535,
-                description: "Port on which the server will listen"
-              ],
-              local_ip: [
-                spec: binary(),
-                default: "127.0.0.1",
+  def_options socket: [
+                spec: port(),
```

Comment:
This probably should have the tcp socket spec. With socket backend for TCP sockets the implementation is no longer using ports and here you will receive some opaque tuple instead.

Reply:
Also we are missing the documentation of what will happen with a socket and that a user will receive a notification that he needs to pass the control, information such as the socket should not be active and that the source will be the first one to request the packet.